clients/src/main/java/org/apache/kafka/clients/producer/ProducerInterceptor.java

```diff
@@ -86,7 +87,32 @@ public interface ProducerInterceptor<K, V> extends Configurable, AutoCloseable {
      *                 {@link org.apache.kafka.clients.producer.KafkaProducer#send(ProducerRecord)}.
      * @param exception The exception thrown during processing of this record. Null if no error occurred.
      */
-    void onAcknowledgement(RecordMetadata metadata, Exception exception);
+    default void onAcknowledgement(RecordMetadata metadata, Exception exception) {};
+
+    /**
+     * This method is called when the record sent to the server has been acknowledged, or when sending the record fails before
+     * it gets sent to the server.
+     * <p>
+     * This method is generally called just before the user callback is called, and in additional cases when <code>KafkaProducer.send()</code>
+     * throws an exception.
+     * <p>
+     * Any exception thrown by this method will be ignored by the caller.
```

Comment:
This can be moved to the end of the Java doc and just before the params definition as a note 
```suggestion
     * Note that any exception thrown by this method will be ignored by the caller.
```